sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/Queue/OperationQueue.cs

```diff
@@ -149,6 +149,32 @@ public Task<IDisposable> LockItemAsync(string id, CancellationToken cancellation
             }
         }
 
+        public virtual async Task<bool> UpdateAsync(string id, long version, JObject item)
+        {
+            try
+            {
+                MobileServiceTableOperation op = await GetOperationAsync(id);
+                if (op == null || op.Version != version)
+                {
+                    return false;
+                }
+
+                // Change the operation state back to pending since this is a newly updated operation without any conflicts
+                op.State = MobileServiceTableOperationState.Pending;
```

Comment:
This can be discussed. Since the related errors have been cleared and operation has been essentially not attempted, it seemed logical to reset it to pending.